baselines/diabetic_retinopathy_detection/radial.py

```diff
@@ -0,0 +1,356 @@
+
+    # Linearly scale learning rate and the decay epochs by vanilla settings.
+    base_lr = FLAGS.base_learning_rate * batch_size / DEFAULT_BATCH_SIZE
+    lr_decay_epochs = [
+        (int(start_epoch_str) * FLAGS.train_epochs) // DEFAULT_NUM_EPOCHS
+        for start_epoch_str in FLAGS.lr_decay_epochs]
+
+    lr_schedule = utils.LearningRateSchedule(
+        steps_per_epoch,
+        base_lr,
+        decay_ratio=FLAGS.lr_decay_ratio,
+        decay_epochs=lr_decay_epochs,
+        warmup_epochs=FLAGS.lr_warmup_epochs)
+    optimizer = tf.keras.optimizers.SGD(
+      lr_schedule, momentum=0.9, nesterov=True)
+    metrics = utils.get_diabetic_retinopathy_base_metrics(
+      use_tpu=use_tpu, num_bins=FLAGS.num_bins)
+    metrics.update({
+      'train/kl': tf.keras.metrics.Mean(),
+      'train/kl_scale': tf.keras.metrics.Mean()})
+    checkpoint = tf.train.Checkpoint(model=model, optimizer=optimizer)
+    latest_checkpoint = tf.train.latest_checkpoint(FLAGS.output_dir)
+    initial_epoch = 0
+    if latest_checkpoint:
+      # checkpoint.restore must be within a strategy.scope()
+      # so that optimizer slot variables are mirrored.
+      checkpoint.restore(latest_checkpoint)
+      logging.info('Loaded checkpoint %s', latest_checkpoint)
+      initial_epoch = optimizer.iterations.numpy() // steps_per_epoch
+
+  # Finally, define OOD metrics outside the accelerator scope for CPU eval.
+  # This will cause an error on TPU.
+  if not use_tpu:
+    metrics.update({
+      'train/auc': tf.keras.metrics.AUC(),
+      'test/auc': tf.keras.metrics.AUC()})
+
+  @tf.function
+  def train_step(iterator):
+    """Training step function."""
+
+    def step_fn(inputs):
+      """Per-replica step function."""
+      images = inputs['features']
+      labels = inputs['labels']
+      with tf.GradientTape() as tape:
+        logits = model(images, training=True)
+        if FLAGS.use_bfloat16:
+          logits = tf.cast(logits, tf.float32)
+
+        negative_log_likelihood = tf.reduce_mean(
+            tf.keras.losses.binary_crossentropy(
+                y_true=tf.expand_dims(labels, axis=-1),
+                y_pred=logits,
+                from_logits=True))
+
+        filtered_variables = []
+        for var in model.trainable_variables:
+          # Apply l2 on the BN parameters and bias terms. This
+          # excludes only fast weight approximate posterior/prior parameters,
+          # but pay caution to their naming scheme.
+          if 'bn' in var.name or 'bias' in var.name:
+            filtered_variables.append(tf.reshape(var, (-1,)))
+
+        l2_loss = FLAGS.l2 * 2 * tf.nn.l2_loss(
+            tf.concat(filtered_variables, axis=0))
+        kl = sum(model.losses)
+        kl_scale = tf.cast(optimizer.iterations + 1, kl.dtype)
+        kl_scale /= steps_per_epoch * FLAGS.kl_annealing_epochs
+        kl_scale = tf.minimum(1., kl_scale)
+        kl_loss = kl_scale * kl
+
+        loss = negative_log_likelihood + l2_loss + kl_loss
```

Comment:
I really don't think this should use an L2 loss. Definitely if you aren't using tied means it should not have l2 loss.

Reply:
I agree this would no longer be principled, raising and explaining this now.